source/CCodeInjector.cpp

```diff
@@ -1,35 +1,40 @@
 #include "stdafx.h"
-#include "cleo.h"
+#include "CleoBase.h"
 #include "CDebug.h"
 #include "CCodeInjector.h"
+#include <windows.h> 
 
 namespace CLEO
 {
     void CCodeInjector::OpenReadWriteAccess()
     {
         if (bAccessOpen) return;
 
-        auto dwLoadOffset = static_cast<memory_pointer>(GetModuleHandle(nullptr));
+        auto dwLoadOffset = reinterpret_cast<memory_pointer>(GetModuleHandle(nullptr));
 
-        // Unprotect image - make .text and .rdata section writeable
-        auto pImageBase = (BYTE *)dwLoadOffset;
-        auto pDosHeader = (PIMAGE_DOS_HEADER)dwLoadOffset;
-        auto pNtHeader = (PIMAGE_NT_HEADERS)(pImageBase + pDosHeader->e_lfanew);
+        // Unprotect image - make .text and .rdata section writable
+        auto pImageBase = reinterpret_cast<BYTE *>(dwLoadOffset);
+        auto pDosHeader = reinterpret_cast<PIMAGE_DOS_HEADER>(dwLoadOffset);
+        auto pNtHeader = reinterpret_cast<PIMAGE_NT_HEADERS>(pImageBase + pDosHeader->e_lfanew);
         auto pSection = IMAGE_FIRST_SECTION(pNtHeader);
 
-        for (int i = pNtHeader->FileHeader.NumberOfSections; i; i--, pSection++)
+        for (int i = 0; i < pNtHeader->FileHeader.NumberOfSections; ++i, ++pSection)
         {
-            if (!strcmp((char*)pSection->Name, ".text") || !strcmp((char*)pSection->Name, ".rdata"))
+            std::string sectionName(reinterpret_cast<char*>(pSection->Name));
```

Comment:
I do not see how it is more safe. Only thing it introduces is additional memory copy.